backend/src/app.ts

```diff
@@ -108,3 +127,33 @@ app.put("/converter/save", updateConverter);
 
 // compiler
 app.post("/compiler/compile", compileCode);
+
+// passport
+app.get("/auth/google", passport.authenticate("google", { scope: ["email", "profile"] }));
+
+app.get(
+    "/auth/google/callback",
+    passport.authenticate("google", {
+        successRedirect: "/auth/success",
+        failureRedirect: "/error",
+    })
+);
+
+app.get("/auth/success", (req: Request, res: Response) => {
+    // ten endpoint ma za zadanie wysłać wiadomość do openera (głównego okna AlgoDebug), dzięki czemu poinformujemy aplikację o poprawnej autoryzacji oraz zamknąć okno z autoryzacją.
+    res.setHeader("Content-Type", "text/html");
+    const user = JSON.stringify(req.user);
+    res.send(
+        `<script>window.onload = () => {if (!window.opener) return; window.opener.postMessage(${user}, "http://localhost:8081"); window.close()}</script>`
+    );
+});
+
+app.get("/logout", (req: Request, res: Response) => {
+    req.logout(() => {
+        res.status(200).json({ loggedIn: false });
+    });
+});
+
+app.get("/auth/verify", (req: Request, res: Response) => {
+    res.status(200).json({ loggedIn: req.isAuthenticated() });
```

Comment:
Aaa, jeszcze jakby tu się dało zwrócić obiekt zalogowanego użytkownika to by było fajnie. Kojarzę, że chyba powinien być gdzieś w tym obiekcie req ale już nie pamiętam dokładnie.